### PR TITLE
Move ComponentSelector from PowerAnalytics to InfrastructureSystems: PSY portion

### PR DESCRIPTION
**Pull request by GabrielKS**
This draft PR will get filled in as @tengis-nrl and I move the ComponentSelector features from PowerAnalytics to InfrastructureSystems.

### CODE REVIEW

**Comment by tengis-nrl** [comment]
image

**Comment by tengis-nrl** [comment]
image

File: src/PowerSystems.jl [hunk 1]
```diff
@@ -439,6 +439,20 @@ export StructDefinition
 export generate_struct_file
 export generate_struct_files
 
+#ComponentSelector
```
Comment (GabrielKS) [inline src/PowerSystems.jl:442]:
```suggestion
# ComponentSelector
```

File: test/test_component_selector.jl [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+test_sys = PSB.build_system(PSB.PSITestSystems, "c_sys5_all_components")
+test_sys2 = PSB.build_system(PSB.PSISystems, "5_bus_hydro_uc_sys")
+
+sort_name(x) = sort(collect(x); by = get_name)
+
+@testset "Test helper functions" begin
+    @test subtype_to_string(ThermalStandard) == "ThermalStandard"
+    @test component_to_qualified_string(ThermalStandard, "Solitude") ==
+          "ThermalStandard__Solitude"
+end
```
Comment (GabrielKS) [inline test/test_component_selector.jl:12]: Why was the third test
```Julia
@test component_to_qualified_string(
        PSY.get_component(ThermalStandard, test_sys, "Solitude"),
    ) == "ThermalStandard__Solitude"
```
removed from here?

Comment (tengis-nrl) [reply test/test_component_selector.jl:12]: This should be added back,

**Comment by GabrielKS** [comment]
Rebased onto `psy4` and tests pass.

**Pull request closed without merging**